test/e2e/framework/nodes_util.go

```diff
@@ -187,6 +201,16 @@ func NodeUpgrade(f *Framework, v string, img string) error {
 	return waitForNodesReadyAfterUpgrade(f)
 }
 
+func nodeUpgradeLocal(v string) error {
+	script := localUpgradeScript()
```

Comment:
Not a fan of this idea.  
/assign @timothysc 
/hold

Reply:
Can you elaborate? I've implemented this in a way that mirrors how upgrades are run in the e2e tests for [GCE](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/nodes_util.go#L78) and [GKE](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/nodes_util.go#L113) clusters, but I'm happy to make changes if you can suggest a better way to accomplish the goal of running upgrade tests against arbitrary clusters.

Reply:
The current process of how upgrades are managed in testing today are cobbled in test-infra today.  

It's much saner todo multi-step process out of the core e2es.

Reply:
Thanks for elaborating, and apologies for my delay in coming back to this (this work has fallen down my stack of in-progress things).

I definitely agree - the upgrade process in the tests feels cobbled together at the moment. I'm happy to help alleviate that as part of or as a prerequisite for this change.

That said, I'm not sure I understand what you mean by "out of the core e2es". Is the idea that the upgrade tests themselves should manage the upgrade process, leaving the core framework out of it? Or are you suggesting that upgrade-related tests move out of the e2e framework entirely?